### PR TITLE
Remove broken ``generate-man`` option

### LINKED ISSUE #1887: Add the man page generation back in
Currently is has been stripped out to make the LongHelp classes work, but it needs to be added back in.

### LINKED ISSUE #5283: Generated man page for pylint are untested and unused
### Question

@sandrotosi, how do you generate the man page for pylint when you release ? Is that handled in your own debian repository for pylint ? After some retro-engineering I realized that the generation functions still exist, but are untested and probably unused (?) 

### Documentation for future user

The option for man page generation should be detailed /tested or removed.

### PR DESCRIPTION
**Pull request by DanielNoord**
- [x] Add yourself to CONTRIBUTORS if you are a new contributor.
- [x] Add a ChangeLog entry describing what your PR does.
- [x] If it's a new feature, or an important bug fix, add a What's New entry in
      `doc/whatsnew/<current release.rst>`.
- [x] Write a good description on what the PR does.

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :bug: Bug fix          |

## Description

Turns out this was broken, probably for quite some time. Since nobody opened an issue about this and nobody seems to sue this anymore, I think this can be considered dead code. It wasn't tested anyway.
Closes #5283
Closes #1887

When you try to use this on `main`:
```console
❯ pylint --generate-man
Traceback (most recent call last):
  File "/Users/daniel/.pyenv/versions/pylint-3.10.0/bin/pylint", line 33, in <module>
    sys.exit(load_entry_point('pylint', 'console_scripts', 'pylint')())
  File "/Users/daniel/DocumentenLaptop/Programming/Github/pylint/pylint/__init__.py", line 22, in run_pylint
    PylintRun(argv or sys.argv[1:])
  File "/Users/daniel/DocumentenLaptop/Programming/Github/pylint/pylint/lint/run.py", line 350, in __init__
    args = _config_initialization(
  File "/Users/daniel/DocumentenLaptop/Programming/Github/pylint/pylint/config/config_initialization.py", line 73, in _config_initialization
    parsed_args_list = linter.load_command_line_configuration(args_list)
  File "/Users/daniel/DocumentenLaptop/Programming/Github/pylint/pylint/config/option_manager_mixin.py", line 338, in load_command_line_configuration
    (options, args) = self.cmdline_parser.parse_args(args=args)
  File "/Users/daniel/.pyenv/versions/3.10.0/lib/python3.10/optparse.py", line 1387, in parse_args
    stop = self._process_args(largs, rargs, values)
  File "/Users/daniel/.pyenv/versions/3.10.0/lib/python3.10/optparse.py", line 1427, in _process_args
    self._process_long_opt(rargs, values)
  File "/Users/daniel/.pyenv/versions/3.10.0/lib/python3.10/optparse.py", line 1501, in _process_long_opt
    option.process(opt, value, values, self)
  File "/Users/daniel/DocumentenLaptop/Programming/Github/pylint/pylint/config/option.py", line 203, in process
    return self.take_action(self.action, self.dest, opt, value, values, parser)
  File "/Users/daniel/.pyenv/versions/3.10.0/lib/python3.10/optparse.py", line 805, in take_action
    self.callback(self, opt, value, parser, *args, **kwargs)
  File "/Users/daniel/DocumentenLaptop/Programming/Github/pylint/pylint/lint/run.py", line 442, in cb_generate_manpage
    self.linter.generate_manpage(__pkginfo__)
  File "/Users/daniel/DocumentenLaptop/Programming/Github/pylint/pylint/config/option_manager_mixin.py", line 230, in generate_manpage
    formatter.format_head(self.cmdline_parser, pkginfo, section),
  File "/Users/daniel/DocumentenLaptop/Programming/Github/pylint/pylint/config/man_help_formatter.py", line 46, in format_head
    short_desc = self.format_short_description(pgm, pkginfo.description)
AttributeError: module 'pylint.__pkginfo__' has no attribute 'description'
```

### CODE REVIEW

File: ChangeLog [hunk 1]
```diff
@@ -14,6 +14,10 @@ Release date: TBA
 
   Closes #6074
 
+* Removed the broken ``generate-man`` option.
+
+  Closes #5283
```
Comment (Pierre-Sassoulas) [inline ChangeLog:19]:
```suggestion
  Closes #5283
  Closes #1887
```

File: doc/whatsnew/2.14.rst [hunk 2]
```diff
@@ -46,6 +46,10 @@ Other Changes
 
   Closes #5931
 
+* Removed the broken ``generate-man`` option.
+
+  Closes #5283
```
Comment (Pierre-Sassoulas) [inline doc/whatsnew/2.14.rst:51]:
```suggestion
  Closes #5283
  Closes #1887
```

**Review by Pierre-Sassoulas — CHANGES_REQUESTED** [review]

File: ChangeLog (hunk 1 above)
Comment (DanielNoord) [reply ChangeLog:19]: Oh wow, this was broken for a very long time 😄

Comment (Pierre-Sassoulas) [reply ChangeLog:19]: And absolutely no one complained, it's all maintainer issues 😄

**Review by Pierre-Sassoulas — APPROVED** [review]
I really like removing unused code.

**Pull request merged**